rclrs/src/lib.rs

```diff
@@ -302,7 +297,11 @@ mod tests {
                     return Err(msg);
                 }
             }
-            Ok(())
         }
+
+        let empty_non_ros_args = extract_non_ros_args(vec![]).unwrap();
```

Comment:
I feel like this should actually be spun into a separate test function - the other two cases are both parsing the same dataset for testing, while this case is parsing a new (empty) dataset.